test/e2e/tinkerbellhardwarecount.go

```diff
@@ -0,0 +1,43 @@
+package e2e
+
+import (
+	_ "embed"
+	"fmt"
+
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed tinkerbell_hardware_count.yaml
+var tinkerbellHardwareCountFile string
+
+// TinkerbellTest maps each Tinkbell test with the hardware count needed for the test.
+type TinkerbellTest struct {
+	Name  string `yaml:"name"`
+	Count int    `yaml:"count"`
+}
+
+// GetTinkerbellHardwareMap returns a map of tink tests with associated hardware count.
+func GetTinkerbellHardwareMap() (map[string]int, error) {
+	tinkerbellHardwareMap, err := readTinkerbellTestFile()
+	if err != nil {
+		return nil, fmt.Errorf("error getting hardware count for tinkerbell tests: %v", err)
+	}
+	return tinkerbellHardwareMap, nil
+}
+
+func readTinkerbellTestFile() (map[string]int, error) {
```

Comment:
Can the body of this function be in `GetTinkerbellHardwareMap()`?

Reply:
We could. but the idea was that the test runner should only be needing the Hardware configuration (which is the exported function) but not necessarily need to know or worry about the logic for parsing which is done by `readTinkerbellTestFile()` within the e2e package.